### ISSUE TITLE
[Bug]: Plugin forgets authentication

### ISSUE THREAD

**Issue by tophee** [XeroxDev/YTMD-StreamDeck#143]
### Consent

- [X] I have searched the existing issues and verified that there is no open issue for the same subject.
- [X] I understand that the YTMDesktop Elgato Plugin has NO affiliation with Google, YouTube or the YTMDesktop application.

### Current Behavior

(0. Remove Stream Deck Connector from list of authorized companions in Youtube Music Desktop App)
1. Press authenticate button
2. Pop-up appears asking to compare code and grant access
3. Plogin shows "Auth Status: Authenticated!"
4. Stream Deck Connector appears as "authorized companion" in Youtube Music Desktop App
5. Pressing the button on streamdeck has no effect
6. When navigating away from the button in Stream Deck app and back again, the authorize button is back (i.e. the plugin is no longer authorized. Or rather: it believes it isn't because it still shows as authorized in Youtube Music Desktop App.)

### Expected Behavior

Same as above, except, of course, that the button will work in step 5.

### Steps To Reproduce

see above

### OS

macOS

### OS Version

15.1.1

### Plugin Version

v2.2.0

### YTMDesktop Version

v2.0.6

### How did you download the software?

Elgato Store

### Anything else?

I almost gave up submitting this report because the submit button was greyed out and I couldn't see why. Turned out that the plugin version appeared to be pre-filled with the correct value but the value needed to be filled in manually.

**Comment by tophee** [comment]
I tried all kinds of restarting and what not but nothing helped. When I then pressed play on the Youtube Music Desktop App, the plugin picked that up and it started to work.

So, in order to pinpoint what exactly was going on, I should probably mention that the button I was working with was the play/pause button. Maybe there is something specific about the play button that needs to be "initialized" by starting the first playback on the app?

Or perhaps this is, after all, a bug in the companion server? Maybe the companion server did actually receive the play command but didn't know what to play because the main app hadn't yet told it (which it did once I pressed play in the app)?

**Comment by XeroxDev** [comment]
Hey! I've tried to reproduce it but to no avail. But will look deeper into it. Maybe I can reproduce it in some other ways.

**Comment by kyle-littlestar** [comment]
Hi! I have this issue with the play/pause button as well. 

The exact steps I took to run into this issue is:

- Open YTM App
- Wait for it to fully load (confirm play button gets a green checkmark on stream deck physical button)
- Hit Play button on stream deck.

This for me, does not start playback even if the app loads in with a playlist ready to go.

My current work around is to open the integration in settings and re-authorize Stream Deck. Not sure if this is on YTM app's side or the SD side. I can confirm that "Stream Deck Connector" is present prior to re-authorizing.

**Comment by XeroxDev** [comment]
Hey @tophee and @kyle-littlestar

If this is really the problem I think and I'm quite sure, a User in my discord and I could reproduce it.

It seems that it's not related to the authentication, nor the functionality of the plugin itself which is why I couldn't reproduce it on my setup. 

The problem is actually within YTMDesktop itself. You can confirm this the following way:

If the problem occurs, that the play button doesn't work (a normal restart of the YTMDesktop app should be sufficient), just try it with your normal play/pause media key on your keyboard. This will, as well, not be able to start playing a track. But if you first click into ytmdesktop (or, to make even more sure, start the entire playback manually once) the media key and the plugins play/pause action will resume to work again.

We're not exactly sure, why that happens and this also has finally to be fixed by the YTMDesktop team.

A workaround for it seems to be the following:

Enabling "continue were you left of" in the YTMDesktop settings and then disabling "pause on application start" (see screenshot of discord conversation)
![Screenshot_20250125_223500_Discord.jpg](https://github.com/user-attachments/assets/7d77f188-00aa-4e5b-9c51-765983b65d26)

This will force the track to start automatically and allows the play/pause action, media key, etc. to work again.

TL;DR this is not a bug within this specific plugin. Any plugin or action sending the play/pause command to YTMDesktop has the same issue. This is, because there's a bug within the YTMDesktop application itself and can not be fixed from outside YTMDesktop.
